src/gui/newwizard/SetupWizardWindow.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <QDialog>
+#include <QEvent>
+#include <QKeyEvent>
+#include <QList>
+#include <QPair>
+#include <QStackedLayout>
+#include <QStackedWidget>
+
+#include "3rdparty/QProgressIndicator/QProgressIndicator.h"
+#include "Pagination.h"
+#include "SetupWizardAccountBuilder.h"
+#include "pages/AbstractSetupWizardPage.h"
+
+namespace Ui {
+class SetupWizardWindow;
+}
+
+namespace OCC::Wizard {
+/**
+     * This class contains the UI-specific code. It hides the complexity from the controller, and provides a high-level API.
+     */
+class SetupWizardWindow : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit SetupWizardWindow(QWidget *parent);
+
+    /**
+         * Set entries in the pagination at the bottom of the wizard UI.
+         * The entries are identified by their position in the list (read: index).
+         */
+    void setPaginationEntries(const QStringList &paginationEntries);
+
+    /**
+     * Render this page within the wizard window.
+     * @param page page to render
+     * @param index index to highlight in pagination (also used to decide which buttons to enable)
+     */
+    void displayPage(AbstractSetupWizardPage *page, PageIndex index);
+
+    void showErrorMessage(const QString &errorMessage);
+
+    void disableNextButton();
+
+protected:
+    bool eventFilter(QObject *obj, QEvent *event) override;
+
+Q_SIGNALS:
+    void paginationEntryClicked(PageIndex currentPage, PageIndex clickedPageIndex);
+    void nextButtonClicked(PageIndex currentPage);
+    void backButtonClicked(PageIndex currentPage);
+
+public Q_SLOTS:
+    /**
+     * Show "transition to next page" animation. Use displayPage(...) to end it.
+     */
+    void slotStartTransition();
+
+private Q_SLOTS:
+    void slotReplaceContent(QWidget *newWidget);
+    void slotHideErrorMessageWidget();
+
+private:
+    QSharedPointer<::Ui::SetupWizardWindow> _ui;
```

Comment:
Please use parenting.
There is no reason to use shared pointers.
Maybe scoped pointers but with Qt there is usually no need for smart pointers, you'll only end up with undefined lifetimes.

Reply:
In this use case, the shared pointer is just used to auto-delete the auto-generated class in the destructor. Since we don't intend to ever implement move behavior, a scoped pointer would work, too. However, scoped pointers require the class declaration to be included, which means including the auto-generated headers in those headers:

```
error: invalid application of ‘sizeof’ to incomplete type ‘Ui::AccountConfiguredWizardPage’
```

I don't think you can use parenting properly with those auto-generated classes, can you? They are not `QObject`s, but plain classes.